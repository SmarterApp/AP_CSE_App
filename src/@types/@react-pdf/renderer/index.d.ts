// tslint:disable:no-any no-unsafe-any
declare module '@react-pdf/renderer' {
  import React from 'react';

  interface StyleSheet {
    create: (Object: any) => any;
  }

  interface DocumentProps {
    title?: string;
    author?: string;
    subject?: string;
    keywords?: string;
    creator?: string;
    producer?: string;
    onRender?: Function;
    shallow?: boolean;
    width?: string | number;
    height?: string | number;
    style?: Object;
    className?: string;
  }
  export const Document: React.SFC<DocumentProps>;

  interface PageProps {
    size?: string | Object | any[];
    orientation?: string;
    wrap?: boolean;
    style?: Object | any[];
    debug?: boolean;
    ruler?: boolean;
    rulerSteps?: number | string;
    verticalRuler?: boolean;
    verticalRulerSteps?: number | string;
    horizontalRuler?: boolean;
    horizontalRulerSteps?: number | string;
  }
  export const Page: React.SFC<PageProps>;

  interface ViewProps {
    wrap?: boolean;
    style?: Object | any[];
    render?: Function;
    debug?: boolean;
    fixed?: boolean;
    minPresenceAhead?: number;
  }
  export const View: React.SFC<ViewProps>;

  interface TextProps {
    wrap?: boolean;
    style?: Object | any[];
    render?: Function;
    debug?: boolean;
    fixed?: boolean;
    minPresenceAhead?: number;
    widows?: number;
    orphans?: number;
  }
  export const Text: React.SFC<TextProps>;

  interface LinkProps {
    wrap?: boolean;
    style?: Object | any[];
    render?: Function;
    debug?: boolean;
    fixed?: boolean;
  }
  export const Link: React.SFC<LinkProps>;

  interface ImageProps {
    src: string;
    style: Object | any[];
    debug?: boolean;
    fixed?: boolean;
  }
  export const Image: React.SFC<ImageProps>;

  interface NoteProps {
    style?: Object | any[];
    children: string;
    fixed?: boolean;
  }
  export const Note: React.SFC<NoteProps>;

  interface PDFViewerProps {
    style?: Object | Object[] | any | any[];
    className?: string;
  }
  export const PDFViewer: React.SFC<PDFViewerProps>;

  interface PDFDownloadLinkProps {
    document: Document;
    filename: string;
    style?: Object | any[];
    classname?: string;
    children: Function | Node;
  }
  export const PDFDownloadLink: React.SFC<PDFDownloadLinkProps>;
  export const StyleSheet: StyleSheet;
}
