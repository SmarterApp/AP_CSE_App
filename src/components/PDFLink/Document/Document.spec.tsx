import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PDFViewer } from '@react-pdf/renderer';

import { CustomDocument } from './';

describe('CustomDocument', () => {
  it(`renders an ELA target`, () => {
    const component = shallow(CustomDocument);

    expect(toJson(component)).toMatchSnapshot();
  });
});
