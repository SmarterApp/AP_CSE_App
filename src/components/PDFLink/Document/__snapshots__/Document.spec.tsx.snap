// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomDocument renders an ELA target 1`] = `
<DOCUMENT>
  <PAGE>
    <TEXT>
      Hello, World
    </TEXT>
  </PAGE>
</DOCUMENT>
`;
